tests/functional/test_config_after_upgrade.py

```diff
@@ -0,0 +1,435 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for cluster and provider configs after upgrade"""
+import json
+from collections import OrderedDict
+from typing import Callable, Union
+
+import allure
+import pytest
+from adcm_client.objects import GroupConfig
+from adcm_pytest_plugin.utils import get_data_dir, ordered_dict_to_dict
+from allure_commons.types import AttachmentType
+from tests.functional.plugin_utils import AnyADCMObject
+
+
+###############################
+# Tools
+###############################
+
+
+def _update_config_and_attr_for_new_field_test(config: OrderedDict, attr: OrderedDict, group_config=False):
```

Comment:
Should you add docstrings and hints for return? 
And in other functions in this file.

Reply:
These are local service functions. I don't think that docstrings will add much here

Reply:
return hints were added